### PR TITLE
feat:changes for user token management

### PR DESCRIPTION
**Pull request by prashbnair**

### CODE REVIEW

**Comment by prashbnair** [comment]
@arajkumar @yzainee Can you please take a quick look? If this looks good, I can make changes to the other repos. The build is failing because nodes do not exist.

**Comment by prashbnair** [comment]
The idea is take out the common code here and then use it in the api-server and stacks-report (for cron job). This library is already pulled in as a dependency in both repos

File: f8a_utils/user_token_utils.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+@retry(reraise=True, stop=tenacity.stop_after_attempt(6),
+       wait=tenacity.wait_exponential(multiplier=2, min=4))
+def is_snyk_token_valid(snyk_api_token):
+    """Validate Snyk API token."""
+    try:
+        response = requests.post(SNYK_API_TOKEN_VALIDATION_URL, json={'api': snyk_api_token})
+        return response.status_code == 200
+    except Exception as e:
+        logger.exception("Encountered exception calling Snyk")
```
Comment (yzainee) [inline f8a_utils/user_token_utils.py:21]: maybe change the text to "Encountered exception while calling Snyk API"

**Review by yzainee — APPROVED** [review]

**Review by arajkumar — COMMENTED** [review]
It might be irrelevant, but how are we going to deal with future encryption key chances?

scenario:
- Service is running with encryption key x
- Chance encryption key to y & deploy
What would be the status of fields which was encrypted using old key x?

**Comment by prashbnair** [comment]
I am not sure I understand the question but we are using the same secret across both services. If the key is to change then all the encrypted text will have to be migrated to use the new key using a one time job and then the new key can be used in both services.

**Comment by prashbnair** [comment]
@arajkumar Please let me know if you have any concerns. If not can you please approve the PR?

**Review by arajkumar — APPROVED** [review]
lgtm

**Pull request merged**